zerver/views/realm.py

```diff
@@ -232,6 +233,10 @@ def update_realm(
     if enable_spectator_access:
         realm.ensure_not_on_limited_plan()
 
+    if can_access_all_users_group_id is not None:
+        # Remove this when the feature is ready for production.
+        assert settings.DEVELOPMENT
```

Comment:
I have used `assert` here instead of raising `JsonableErrror` since we do the same for demo orgs feature.